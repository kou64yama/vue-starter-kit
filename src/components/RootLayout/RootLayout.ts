import Vue from 'vue';
import AppHeader from '../AppHeader';
import AppFeedback from '../AppFeedback';
import AppFooter from '../AppFooter';

export default Vue.extend({
  components: {
    AppHeader,
    AppFeedback,
    AppFooter,
  },
});
