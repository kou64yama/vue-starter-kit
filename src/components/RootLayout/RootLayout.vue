<template>
  <div>
    <app-header />
    <router-view />
    <app-feedback />
    <app-footer />
  </div>
</template>

<script lang="ts" src="./RootLayout.ts"></script>
<style src="./RootLayout.css"></style>
