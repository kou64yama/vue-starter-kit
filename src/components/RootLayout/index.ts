import RootLayout from './RootLayout.vue';

export default RootLayout;
