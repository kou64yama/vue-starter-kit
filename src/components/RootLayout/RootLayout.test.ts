import { shallowMount } from '@vue/test-utils';
import RootLayout from './RootLayout.vue';
import stubs from '@/test/stubs';

describe('RootLayout', () => {
  test('renders children correctly', () => {
    const wrapper = shallowMount(RootLayout, {
      stubs,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
