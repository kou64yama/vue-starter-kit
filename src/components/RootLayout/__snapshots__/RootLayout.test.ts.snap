// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout renders children correctly 1`] = `
<div>
  <appheader-stub />
   
  <routerview-stub />
   
  <appfeedback-stub />
   
  <appfooter-stub />
</div>
`;
